### PR TITLE
roachtest: improve fixtures process

### PR DESCRIPTION
**Pull request by tbg**
- roachtest: remove bogus fixtures (cc @j-low)
- roachtest: make fixture creation less error-prone
- roachtest: update version map and fixtures for 21.1.0

cc @dt @jlinder

### CODE REVIEW

File: pkg/cmd/roachtest/versionupgrade.go [hunk 1]
```diff
@@ -93,13 +93,16 @@ func runVersionUpgrade(ctx context.Context, t *test, c *cluster, buildVersion ve
...
-		makeVersionFixtureAndFatal(ctx, t, c, predV, newV)
+		// empty string which uses the local cockroach binary. Make sure that
+		// this binary then has the correct version. For example, to make a
+		// "v20.2" fixture, you will need a binary that has "v20.2" in the
+		// output of `./cockroach version`, and this process will end up
+		// creating fixtures that have "v20.2" in them. This would be part
+		// of tagging the master branch as v21.1 in the process of going
+		// through the major release for v20.2.
+		var makeFixtureVersion string
+		// makeFixtureVersion = "21.2.0"
```
Comment (otan) [inline pkg/cmd/roachtest/versionupgrade.go:104]: nit: do we still need this?

File: pkg/cmd/roachtest/versionupgrade.go [hunk 2]
```diff
@@ -511,9 +514,33 @@ func stmtFeatureTest(
 // the artifacts. The test will fail on purpose when it's done with instructions
 // on where to move the files.
```
Comment (otan) [inline pkg/cmd/roachtest/versionupgrade.go:515]: can we detail the makeFixtureVersion common case?

**Review by otan — APPROVED** [review]

File: pkg/cmd/roachtest/versionupgrade.go (hunk 1 above)
Comment (tbg) [reply pkg/cmd/roachtest/versionupgrade.go:104]: We do - this should be the common case. I clarified.

File: pkg/cmd/roachtest/versionupgrade.go (hunk 2 above)
Comment (tbg) [reply pkg/cmd/roachtest/versionupgrade.go:515]: not exactly sure what you mean, but I rewrote the comment. Hope that helped :-)

**Comment by tbg** [comment]
bors r=otan

**Pull request merged**